src/arch/x86_64/kernel/processor.rs

```diff
@@ -388,6 +390,23 @@ impl CpuFrequency {
 		Err(())
 	}
 
+	fn detect_from_fdt(&mut self) -> Result<(), ()> {
+		fn mhz_from_fdt() -> Option<NonZero<u16>> {
+			let khz = env::fdt()?
+				.find_node("/hermit,tsc")?
+				.property("khz")?
+				.as_usize()?;
+			let khz = u32::try_from(khz).ok()?;
+			let mhz = u16::try_from(khz / 1000).ok()?;
+			NonZero::new(mhz)
+		}
+
+		let mhz = mhz_from_fdt().ok_or(())?;
+		self.set_detected_cpu_frequency(mhz.get(), CpuFrequencySources::Fdt)?;
+
+		Ok(())
```

Comment:
The other functions in this file are written like
```suggestion
		return self.set_detected_cpu_frequency(mhz.get(), CpuFrequencySources::Fdt);
```

Reply:
Yes, but I consider returning a result instead of handling it and repackaging it to be bad style. :D

I just did not want to trick myself into rewriting this today.